olive/azureml/azureml_client.py

```diff
@@ -98,22 +98,17 @@ def _get_credentials(self):
         2. DefaultAzureCredential
         3. InteractiveBrowserCredential
         """
-        from azure.identity import AzureCliCredential, DefaultAzureCredential, InteractiveBrowserCredential
+        from azure.identity import DefaultAzureCredential, InteractiveBrowserCredential
 
         logger.debug("Getting credentials for MLClient")
         try:
-            credential = AzureCliCredential()
+            credential = DefaultAzureCredential()
```

Comment:
nit: if we'd like to by pass managed identity, we could pass argument to Default one by disable it.

Reply:
Sounds good to me. Will add it in a separated PR.